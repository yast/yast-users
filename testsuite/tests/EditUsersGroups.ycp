/**
 * File:	EditUsersGroups.ycp
 * Module:	Users configurator
 * Summary:	Testing adding and removing user from groups (see bug #136267)
 * Author:	Jiri Suchomel <jsuchome@suse.cz>
 * $Id$
 */

{
    // testedfiles: Users.pm

    import "Directory";
    import "Mode";
    import "Users";

    string tmpdir	= Directory::tmpdir;
    foreach (string file, [ "passwd", "group", "shadow" ], {
	string cmd	= sformat ("/bin/cp ./%1 %2/", file, tmpdir);
	SCR::Execute (.target.bash_output, cmd);
    });
    Users::SetBaseDirectory (tmpdir);
    Users::ReadLocal ();

    map READ = $[
	"etc" : $[
	    "fstab": [],
	    "cryptotab": [],
	    "default": $[
		"useradd": $[
		    "home":	"/home",
		    "groups":	"audio,video",
		    "expire":	nil,
		    "group":	100
		]
	    ],
	],
	"target": $[
	    "stat"	: $[],
	    "size"	: -1,
	    "tmpdir"	: "/tmp/YaST"
	],
	"product": $[
		"features": $[
		    "USE_DESKTOP_SCHEDULER"	: "no",
		    "IO_SCHEDULER"		: "",
		    "ENABLE_AUTOLOGIN"		: "false",
		    "UI_MODE"			: "simple",
		    "EVMS_CONFIG"		: "no",
		    "INCOMPLETE_TRANSLATION_TRESHOLD"	: "99",
		]
	],
	"anyxml"	: nil,
    ];
    map WRITE = $[];
    map EXEC = $[
	"passwd" : $[
	    "init"	: true
	],
	"target" : $[
	    "bash"		: 0,
	    "bash_output"	: $[
		"stdout": "",
	    ],
	],
    ];

    import "Testsuite";

    Mode::SetTest ("test");

    Testsuite::Test (``(Users::Read ()), [READ, WRITE, EXEC], 0);

    map <string,any> group = $[
	"gidNumber"	: 555,
	"cn"		: "ggl",
	"userlist"	: $[ "hh" : 1 ],
	"password"	:"x",
	"type"		: "local",
	"what"		: "add_group"
    ];

    Testsuite::Test (``(Users::AddGroup (group)), [READ, WRITE, EXEC], 0);

    EXEC ["target", "bash_output", "stdout"] = "ggl";

    Testsuite::Test(``(Users::CheckGroup ($[])), [READ, WRITE, EXEC], 0);

    Testsuite::Test(``(Users::CommitGroup ()), [READ, WRITE, EXEC], 0);

    // for home directory checks
    READ["target","stat","isdir"]	= true;

    Users::SelectUserByName ("hh");

    map<string,any> changes = $[
	"grouplist"	: $[
	    "audio"	: 1,
	]
    ];

    Testsuite::Test (``(Users::EditUser (changes)), [READ, WRITE, EXEC], 0);

    EXEC ["target", "bash_output", "stdout"] = "hh";

    string error = (string) Testsuite::Test(``(Users::CheckUser ($[])), [READ,WRITE,EXEC], 0);

    Testsuite::Dump (sformat ("---- unchanged group 'ggl':\n %1", Users::GetGroupByName ("ggl", "")));

    Testsuite::Dump (sformat ("---- unchanged group 'audio':\n %1", Users::GetGroupByName ("audio", "")));

    Testsuite::Test (``(Users::CommitUser ()), [READ, WRITE, EXEC], 0);

    Testsuite::Dump (sformat ("---- changed group 'ggl':\n %1", Users::GetGroupByName ("ggl", "")));

    Testsuite::Dump (sformat ("---- changed group 'audio':\n %1", Users::GetGroupByName ("audio", "")));
}
