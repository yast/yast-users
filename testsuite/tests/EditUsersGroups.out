Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Return	
Return	
Execute	.target.bash_output "echo 'ggl' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"ggl"]
Return	
Return	true
Read	.anyxml "/etc/security/pam_mount.conf.xml" nil
Return	
Execute	.target.bash_output "echo 'hh' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"hh"]
Return	
Dump	---- unchanged group 'ggl':
Dump	 $["cn":"ggl", "gidNumber":"555", "modified":"added", "org_cn":"ggl", "org_gidNumber":"555", "password":"x", "plugins":[], "type":"local", "userlist":$["hh":"1"], "what":"add_group"]
Dump	---- unchanged group 'audio':
Dump	 $["cn":"audio", "gidNumber":"17", "more_users":$[], "type":"system", "userPassword":"x", "userlist":$["ii":"1"]]
Return	true
Dump	---- changed group 'ggl':
Dump	 $["cn":"ggl", "gidNumber":"555", "modified":"added", "org_cn":"ggl", "org_gidNumber":"555", "password":"x", "plugins":[], "type":"local", "userlist":$[], "what":"user_change"]
Dump	---- changed group 'audio':
Dump	 $["cn":"audio", "gidNumber":"17", "modified":"edited", "more_users":$[], "org_cn":"audio", "org_gidNumber":"17", "type":"system", "userPassword":"x", "userlist":$["hh":"1", "ii":"1"], "what":"user_change"]
